subprojects/groovy-jsr223/src/main/java/org/codehaus/groovy/jsr223/GroovyScriptEngineImpl.java

```diff
@@ -427,7 +427,7 @@ private Object callGlobal(String name, Object[] args, ScriptContext ctx) {
     }
 
     // generate a unique name for top-level Script classes
-    private synchronized String generateScriptName() {
+    private synchronized static String generateScriptName() {
```

Comment:
This changes the method from being accessed by one thread per instance to one thread per class.  May not be an issue given what this method does, but in general I would probably not make this kind of bulk change on any `synchronized` method.